src/Infobip.Api.SDK/Email/EmailEndpoints.cs

```diff
@@ -0,0 +1,400 @@
+
+            var serializedPayload = JsonConvert.SerializeObject(requestPayload);
+
+            var url = "email/1/bulks"
+                .AddToQueryString(
+                    new Dictionary<string, object>
+                    {
+                        { "bulkId", bulkId }
+                    }
+                );
+
+            using (var request = new HttpRequestMessage(HttpMethod.Put, url))
+            {
+                request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                request.Content = new StringContent(serializedPayload);
+                request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+
+                using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken))
+                {
+                    await response.ThrowIfRequestWasUnsuccessful();
+
+                    var stream = await response.Content.ReadAsStreamAsync();
+                    return stream.ReadAndDeserializeFromJson<RescheduleEmailMessagesResponse>();
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task<GetSentEmailBulksStatusResponse> GetSentEmailBulksStatus(string bulkId, CancellationToken cancellationToken = default)
+        {
+            var url = "email/1/bulks/status"
+                .AddToQueryString(
+                    new Dictionary<string, object>
+                    {
+                        { "bulkId", bulkId }
+                    }
+                );
+
+            using (var request = new HttpRequestMessage(HttpMethod.Get, url))
+            {
+                request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken))
+                {
+                    await response.ThrowIfRequestWasUnsuccessful();
+
+                    var stream = await response.Content.ReadAsStreamAsync();
+                    return stream.ReadAndDeserializeFromJson<GetSentEmailBulksStatusResponse>();
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task<UpdateScheduledEmailMessagesStatusResponse> UpdateScheduledEmailMessagesStatus(string bulkId, UpdateScheduledEmailMessagesStatusRequest requestPayload, CancellationToken cancellationToken = default)
+        {
+            _requestValidator.Validate(requestPayload);
+
+            var serializedPayload = JsonConvert.SerializeObject(requestPayload);
+
+            var url = "email/1/bulks/status"
+                .AddToQueryString(
+                    new Dictionary<string, object>
+                    {
+                        { "bulkId", bulkId }
+                    }
+                );
+
+            using (var request = new HttpRequestMessage(HttpMethod.Put, url))
```

Comment:
I just took a look at this email endpoint. Creation of `HttpRequestMessage` is something that we could have create to be abstract and generic. I suppose all other channels use same principle. **We will not change it now**. I can do that refactoring assuming all endpoints have functionality. Just making a note.